core/units.h

```diff
@@ -96,6 +100,7 @@ typedef struct
 typedef struct
 {
 	int16_t degrees;
+	int valid;
 } bearing_t; // compass bearing
 
 typedef struct
```

Comment:
Not a fan of these 'valid' members. Simply set these to a known invalid value (and create a macro DURATION_VALID() and BEARING_VALID() that checks for that value).

Reply:
Ok for me and I also won't do this one in a hurry.
And I please need some guidance for a better solution.

In general I would prefer to keep a value '0' being the "flag" for not initialized as default because this works most of the time.
For some items like bearing or ndl we need s.th. else. I would prefer '-1'. Would it be ok to change duration_t from uint to int?
And using '-1' (or anything else != 0) also means that we need to take care to init this in some places. Currently we use e.g. a calloc to init the memory for a new sample to all '0'. We would need to set some specific '-1' then. But don't worry, that's not more than a few lines of code. On the other hand '-1' will make the data transfer from samples to plot_data easier again.